airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/DefaultAirbyteDestination.java

```diff
@@ -138,7 +138,16 @@ public boolean isFinished() {
       return false;
     }
 
-    return !destinationProcess.isAlive();
+    // Also short circuit here to avoid checking the exit value until the process has actually finished.
+    if (destinationProcess.isAlive()) {
+      return false;
+    }
+
+    final int exitValue = destinationProcess.exitValue();
```

Comment:
Can we cache this so we don't pull `exitValue` more than once even if we call `isFinished` multiple times?